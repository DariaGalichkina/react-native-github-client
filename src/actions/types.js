/**
 * Created by Pavel_Ryabichenko on 21.07.2017.
 */
export const LOGIN_CHANGED = 'login_changed';
export const PASSWORD_CHANGED = 'password_changed';
export const LOGIN_USER_SUCCESS = 'login_user_success';
export const LOGIN_USER_FAIL = 'login_user_fail';
export const LOGIN_USER = 'login_user';

export const REPOSITORIES_FETCH = 'repositories_fetch';
export const REPOSITORIES_FETCH_SUCCESS = 'repositories_fetch_success';
export const REPOSITORIES_FETCH_FAIL = 'repositories_fetch_fail';