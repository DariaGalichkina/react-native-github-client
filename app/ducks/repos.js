
import apiProvider from '../lib/apiProvider.js';

const SET_DATA = 'repos/SET_DATA';

export default function reducer(state = null, action) {

  switch (action.type) {

    case SET_DATA:
      return action.reposData;

    default:
      return state;

  }

}

export function getRepos() {

  return async (dispatch) => {
    const reposData = await apiProvider.getRepos();

    dispatch(setReposData(reposData));
  };
}

export function setReposData(reposData) {
  return { type: SET_DATA, reposData };
}
