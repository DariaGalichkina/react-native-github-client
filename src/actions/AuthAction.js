/**
 * Created by Pavel_Ryabichenko on 21.07.2017.
 */

import axios from 'axios';
import {
    LOGIN_CHANGED,
    PASSWORD_CHANGED,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_FAIL,
    LOGIN_USER
} from './types';

export const loginChanged = (text) => {
    return {
        type: LOGIN_CHANGED,
        payload: text
    };
};

export const passwordChanged = (text) => {
    return {
        type: PASSWORD_CHANGED,
        payload: text
    };
};

export const loginUser = ({ login, password }) => {
    return (dispatch) => {
        dispatch({ type: LOGIN_USER });
        axios({
            url: 'https://api.github.com/user',
            //url: './src/reducers/GiHubLoginData.json',
            auth: {
                username: login,
                password: password
            }
        }).then(userResponse =>
            loginUserSuccess(dispatch, userResponse))
          .catch((error) => {
            console.log(error);
            loginUserFail(dispatch);
          });
    };
};

const loginUserFail = (dispatch) => {
    dispatch({
        type: LOGIN_USER_FAIL
    });
};

const loginUserSuccess = (dispatch, userResponse) => {
    dispatch({
        type: LOGIN_USER_SUCCESS,
        payload: userResponse.data
    });
};
