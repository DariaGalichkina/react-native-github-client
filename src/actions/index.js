export * from './AuthAction';
export * from './RepositoriesActions';