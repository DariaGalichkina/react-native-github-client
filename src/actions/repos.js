import GitHubAPI from '../storage/GitHubAPI';

export function reposIsLoading(bool) {
    return {
        type: 'REPOS_IS_LOADING',
        isLoading: bool
    };
}

export function reposHasErrored(bool) {
    return {
        type: 'REPOS_HAS_ERRORED',
        hasErrored: bool
    }
}

export function reposFetchDataSuccess(repos) {
    return {
        type: 'REPOS_FETCH_DATA_SUCCESS',
        repos
    }
}

export function reposFetchData(url) {
    return async (dispatch) => {
        dispatch(reposIsLoading(true));

        var repos = await GitHubAPI.doApiCall('/user/repos');
        console.log(repos);
        if(repos) {
            dispatch(reposFetchDataSuccess(repos));
            dispatch(reposIsLoading(false));
        } else {
            dispatch(reposHasErrored(true));
            dispatch(reposIsLoading(false));
        }

    }
}