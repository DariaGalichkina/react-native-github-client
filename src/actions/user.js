export function userIsLoading(bool) {
    return {
        type: 'USER_IS_LOADING',
        isLoading: bool
    };
}

export function userHasErrored(bool) {
    return {
        type: 'USER_HAS_ERRORED',
        hasErrored: bool
    }
}

export function userLoginSuccess(repos) {
    return {
        type: 'USER_LOGIN_SUCCESS',
        repos
    }
}

export function userLogin(username, password) {
    return async (dispatch) => {
        dispatch(userIsLoading(true));
        var auth = await login(username, password);
        if(auth) {
            dispatch(userLoginSuccess(auth));
            dispatch(userIsLoading(false));
        } else {
            dispatch(userHasErrored(true));
            dispatch(userIsLoading(false));
        }        
    }
}