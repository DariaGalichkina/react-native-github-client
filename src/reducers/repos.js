export function reposHasErrored(state = false, action) {
    switch (action.type) {
        case 'REPOS_HAS_ERRORED':
            return action.hasErrored;

        default:
            return false;
    }
}

export function reposIsLoading(state = false, action) {
    switch (action.type) {
        case 'REPOS_IS_LOADING':
            return action.isLoading;

        default:
            return false;
    }
}

export function repos(state = [], action) {
    switch (action.type) {
        case 'REPOS_FETCH_DATA_SUCCESS':
            return action.repos;

        default:
            return state;
    }
}