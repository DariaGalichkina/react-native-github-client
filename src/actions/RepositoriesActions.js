import axios from 'axios';
import { REPOSITORIES_FETCH,
    REPOSITORIES_FETCH_SUCCESS,
    REPOSITORIES_FETCH_FAIL
} from './types';
import repos from '../data/GitHubRepos.json';

export const repositoriesFetch = (currentUser) => {
    return (dispatch) => {
        dispatch({ type: REPOSITORIES_FETCH });
        if(currentUser && currentUser.repos_url != ''){
            axios({
                url: currentUser.repos_url
            }).then(response => {
                dispatch({ type: REPOSITORIES_FETCH_SUCCESS, payload: response.data });
            }).catch(error => {
                dispatch({ type: REPOSITORIES_FETCH_SUCCESS, payload: repos });
                //dispatch({ type: REPOSITORIES_FETCH_FAIL, payload: error.message });
            });
        } else{
            dispatch({ type: REPOSITORIES_FETCH_FAIL, payload: error.message });
        }
    };
};