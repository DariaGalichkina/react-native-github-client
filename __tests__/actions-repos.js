import 'react-native';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import nock from 'nock';
import thunk from 'redux-thunk';
import { reposFetchData } from '../src/actions/repos';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)


describe('Repos async - reposFetchData', () => {
    afterEach(() => {
        nock.cleanAll()
    })

    it('creates REPOS_FETCH_DATA_SUCCESS when fetching repos has been done', () => {
        nock('https://api.github.com')
          .get('/user/repos')
          .reply(200, [{ name: 'testrepo', stargazers_count: 5 }]);

        const store = mockStore({ repos: [] });

        store.dispatch(reposFetchData('/user/repos'));

        expect(store.getActions()).toMatchSnapshot();

    });
});