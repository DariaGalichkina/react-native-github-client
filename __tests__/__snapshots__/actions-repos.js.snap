// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos async - reposFetchData creates REPOS_FETCH_DATA_SUCCESS when fetching repos has been done 1`] = `
Array [
  Object {
    "isLoading": true,
    "type": "REPOS_IS_LOADING",
  },
]
`;
